js/ui/toolbar/ui.toolbar.base.js

```diff
@@ -343,6 +356,9 @@ var ToolbarBase = CollectionWidget.inherit({
             case "renderAs":
                 this._invalidate();
                 break;
+            case "compactMode":
+                this._dimensionChanged();
```

Comment:
Can we call `this._applyCompactMode()` only without `arrangeItems`?

Reply:
yes, thanks